lib/msal-common/src/utils/Constants.ts

```diff
@@ -401,3 +401,38 @@ export enum JsonTypes {
     Jwt = "JWT",
     Jwk = "JWK"
 }
+
+export enum SilentTokenRetrievalStrategy {
+    /*
+     * acquireTokenSilent will first look in the cache before going to the network with the existing refresh token.
+     * The refresh token will be renewed if it is expired.
+     * (existing default behavior)
+     */
+    CacheAndNetwork = 0, // 0 is falsy, is equivalent to not passing in a SilentTokenRetrievalStrategy
+    /*
+     * acquireTokenSilent will only look in the cache, and will not go to the network with the existing refresh token.
+     * The refresh token will not be renewed if it is expired.
+     */
+    CacheOnly = 1,
+    /*
+     * acquireTokenSilent will first look in the cache before going to the network with the existing refresh token.
+     * The refresh token will not be renewed if it is expired.
+     */
+    CacheOrExistingRefreshToken = 2,
+    /*
+     * acquireTokenSilent will only go to network with the existing refresh token, and will not look in the cache.
+     * The refresh token will not be renewed if it is expired.
+     */
+    NetworkWithExistingRefreshTokenOnly = 3,
+    /*
+     * acquireTokenSilent will only go to network with the existing refresh token, and will not look in the cache.
+     * The refresh token not be renewed if it is expired.
+     * (existing forceRefresh=true behavior)
+     */
+    NetworkWithRefreshToken = 4,
+    /*
+     * acquireTokenSilent will only go to network, and not look in the cache.
```

Comment:
AcquireTokenSilent is an API name that maps to the `refresh_token` flow. If you don't have a refresh token, what will this do?

Reply:
Refresh token flow is only one of 3 parts that `acquireTokenSilent` performs. 1. Cache 2. refresh_token flow 3. Hidden iframe renewal of all tokens